modules/apps/27-interchain-accounts/types/codec_test.go

```diff
@@ -156,3 +307,172 @@ func (suite *TypesTestSuite) TestDeserializeAndSerializeCosmosTxWithAmino() {
 	suite.Require().Error(err)
 	suite.Require().Empty(bz)
 }
+
+func (suite *TypesTestSuite) TestJSONDeserializeCosmosTx() {
+	var cwBytes []byte
+	var protoMessages []proto.Message
+	testCases := []struct {
+		name     string
+		malleate func()
+		expPass  bool
```

Comment:
It might be nice to make this be `expError` and then do `errorsmod.IsOf()`, see #3778

Reply:
I've added this to every test in `codec_test.go`. I think it is not suitable in `relay_test.go` because there are too many places you can get errors in (including the actual meaning of the transactions being executed on chainB). I could still try to add it (with some limitations) though if you really want this.